pkg/codegen/go/gen.go

```diff
@@ -1429,7 +1429,7 @@ func (pkg *pkgContext) genInputTypes(w io.Writer, t *schema.ObjectType, details
 	}
 
 	// Generate the array input.
-	if details.arrayInput {
+	if details.arrayInput && !pkg.names.Has(name+"Array") {
```

Comment:
Reasoning out loud, so worst it can do here unintentionally is drop the helper arrayInput type that was relied on before, but it looks like it only does it if there's an actual conflict. This sounds reasonable! these arrayinput helper types are not quite used as much as normal resource types so this seems good.